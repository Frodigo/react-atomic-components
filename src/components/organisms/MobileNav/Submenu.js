import React, { useState } from 'react';
import PropTypes from 'prop-types';

import Link from '../../atoms/Link';
import IconWithText from '../../molecues/IconWithText';
import Heading from '../../atoms/Text/Heading';

import classes from './MobileNav.module.css';

const Submenu = (props) => {
    const { item } = props;
    const [isSubmenuOpen, setIsSubmenuOpen] = useState(false);

    const handleSubmenu = () => {
        setIsSubmenuOpen(!isSubmenuOpen);
    };

    const submenuItems = item.submenu.map((submenuItem) => (
        <a
            href={submenuItem.url}
            key={submenuItem.id}
            className={classes.submenuItem}
        >
            <IconWithText label={submenuItem.label} image={submenuItem.image} />
        </a>
    ));

    return (
        <>
            <button
                type="button"
                onClick={handleSubmenu}
                className={classes.submenuTrigger}
            >
                <IconWithText
                    label={item.label}
                    image={item.image}
                    isActive={isSubmenuOpen}
                />
            </button>

            <nav
                className={
                    isSubmenuOpen ? classes.submenuOpened : classes.submenu
                }
            >
                <header className={classes.submenuHeader}>
                    <Heading classes="inverse" type="h2">
                        Read about:
                    </Heading>
                </header>

                <div className={classes.submenuItems}>{submenuItems}</div>

                <Link
                    variant="tiertary"
                    href={item.url}
                    style={{ marginTop: '30px' }}
                >
                    Show all
                </Link>
            </nav>
        </>
    );
};

Submenu.propTypes = {
    item: PropTypes.object.isRequired
};

export default Submenu;
