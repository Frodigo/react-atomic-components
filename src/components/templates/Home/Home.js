import React from 'react';

import PageHeader from '../../organisms/PageHeader';
import Hero from '../../molecues/Hero'
import Tile from '../../molecues/Tile';
import MediaBlock from '../../molecues/MediaBlock';
import ContentBlock from '../../molecues/ContentBlock';
import Heading from '../../atoms/Text/Heading';
import Link from '../../atoms/Link';
import Tout from '../../molecues/Tout';

import classes from './Home.module.css';
import FooterMock from "../partials/FooterMock";
import AboutTheAuthor from "../../organisms/AboutTheAuthor";

const Home = (props) => {
    const { ...rest } = props;

    const aboutTheAuthorProps = {
        heading: 'About me',
        name: 'John',
        image: 'https://via.placeholder.com/270x270',
        description:
            'Nam porttitor blandit accumsan. Ut vel dictum sem, a pretium dui. In malesuada enim in dolor euismod, id commodo mi consectetur. Curabitur at vestibulum nisi. Nullam vehicula nisi velit. Mauris turpis nisl, molestie ut ipsum et, suscipit vehicula odio. Vestibulum interdum vestibulum felis ac molestie.',
        ctaUrl: '#',
        ctaLabel: 'Read more'
    };

    const pageHeaderNavItems = [
        {
            url: '#',
            label: 'Home',
            isActive: true
        },
        {
            url: '#',
            label: 'About me'
        },
        {
            url: '#',
            label: 'Blog'
        },
        {
            url: '#',
            label: 'Get in touch'
        }
    ];

    return (
        <div className={classes.Home} {...rest}>
            <PageHeader navItems={pageHeaderNavItems} logoUrl="https://via.placeholder.com/100x30" logoAlt="lorem ipsum" />
            <Hero heading="Lorem ipsum" image={{alt: 'figure', filename: 'https://via.placeholder.com/400x400'}} paragraph="Lorem ipsum dolor sit amet"/>
            <Tile heading="What I do?" />
            <div className={classes.mediaBlockList}>
                <MediaBlock
                    imageUrl="https://via.placeholder.com/290x240"
                    linkUrl="#"
                    text="Nam porttitor blandit accumsan. Ut vel dictum sem, a pretium dui. In malesuada enim in dolor euismod, id commodo mi consectetur. "
                    title="Lorem Ipsum"
                    imageWidth="290"
                    imageHeight="240"
                />
                <MediaBlock
                    imageUrl="https://via.placeholder.com/290x240"
                    linkUrl="#"
                    text="Nam porttitor blandit accumsan. Ut vel dictum sem, a pretium dui. In malesuada enim in dolor euismod, id commodo mi consectetur. "
                    title="Lorem Ipsum"
                    imageWidth="290"
                    imageHeight="240"
                />
            </div>
            <div className={classes.callToActionSection}>
                <ContentBlock
                    heading="Heading level 1"
                    paragraph="Nam porttitor blandit accumsan. Ut vel dictum sem, a pretium dui. In malesuada enim in dolor euismod, id commodo mi consectetur. Curabitur at vestibulum nisi. Nullam vehicula nisi velit. Mauris turpis nisl, molestie ut ipsum et, suscipit vehicula odio. Vestibulum interdum vestibulum felis ac molestie"
                    cta={[{
                        href: '#',
                        children: 'Call to action',
                        variant: 'primary'
                    }]}
                />
            </div>
            <div className={classes.articles}>
                <header className={classes.articlesHeader}>
                    <Heading type="h4">Recent articles</Heading>
                </header>
                <div className={classes.articlesInner}>
                    <Tout
                        imageUrl="https://via.placeholder.com/290x240"
                        linkUrl="#"
                        title="Lorem ipsum"
                        imageWidth="290"
                        imageHeight="240"
                    />
                    <Tout
                        imageUrl="https://via.placeholder.com/290x240"
                        linkUrl="#"
                        title="Lorem ipsum"
                        imageWidth="290"
                        imageHeight="240"
                    />
                </div>
                <footer className={classes.articlesFooter}>
                    <Link href="#" variant="secondary">
                        Show all
                    </Link>
                </footer>
            </div>
            <AboutTheAuthor {...aboutTheAuthorProps} />
            <FooterMock/>
        </div>
    );
};

export default Home;
