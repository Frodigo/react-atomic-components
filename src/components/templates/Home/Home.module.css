.hero {
    position: relative;
    padding: var(--spacing-xl) var(--spacing-md);
}

.hero span {
    display: block;
}

.figure,
.figureDesktop {
    bottom: 15px;
    position: absolute;
    right: 15px;
}

.figureDesktop {
    display: none;
}

.marked {
    color: var(--color-secondary);
}

.callToActionSection {
    background-color: #d8d8d8;
}

.articles {
    padding: var(--spacing-xxl) 0 0 ;
}


.articlesHeader,
.articlesFooter {
    padding: 0 var(--spacing-md);
    margin-bottom: 40px;
    text-align: center;
}

.articlesFooter {
    margin-top: 40px;
}

@media (min-width: 768px) {
    .figure {
        display: none;
    }

    .figureDesktop {
        display: block;
        transform: scale(0.5);
        transform-origin: 100% 100%;
    }

    .mediaBlockList {
        padding-bottom: var(--spacing-xxl);
    }

    .articlesInner {
        display: flex;
    }

    .articlesInner > div {
        flex: 1 1 50%;
        margin: 0 var(--spacing-md);
    }
}

.followMe {
    border-top: 2px solid var(--color-text);
    background-color:#d8d8d8;

}

@media (min-width: 1024px) {
    .figureDesktop {
        transform: scale(0.8);
    }
}

@media (min-width: 1300px) {
    .hero,
    .mediaBlockList,
    .articles {
        margin: 0 auto;
        max-width: var(--desktop-container-width);
        padding: 60px 0;
    }

    .figureDesktop {
        transform: scale(1);
    }
}