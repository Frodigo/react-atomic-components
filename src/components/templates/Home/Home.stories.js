import React from 'react';

import { default as HomeComponent } from './Home';

export default {
    title: 'Templates',
    component: HomeComponent
};

const Template = (args) => <HomeComponent {...args} />;

export const Home = Template.bind({});
