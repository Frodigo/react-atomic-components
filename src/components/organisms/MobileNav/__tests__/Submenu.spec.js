import React from 'react';
import { render } from '@testing-library/react';

import Submenu from '../Submenu';
import Home from '../../../../public/assets/icons/home.svg';

describe('Submenu should be rendered', () => {
    const mockedSubmenu = {
        label: 'Blog',
        image: {
            alt: 'figure',
            url: "image_url"
        },
        url: '/blog',
        id: 2,
        submenu: [
            {
                label: 'PWA Studio',
                image: {
                    alt: 'figure',
                    url: "image_url"
                },
                url: '/pwastudio',
                id: 3
            }
        ]
    };

    test('Submenu renders correctly', () => {
        const { container } = render(<Submenu item={mockedSubmenu} />);

        expect(container).toMatchSnapshot();
    });
});
