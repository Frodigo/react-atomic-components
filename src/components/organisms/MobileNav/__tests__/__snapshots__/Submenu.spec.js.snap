// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Submenu should be rendered Submenu renders correctly 1`] = `
<div>
  <button
    class="submenuTrigger"
    type="button"
  >
    <span
      class="iconContainer"
    >
      <img
        alt="Blog"
        src="[object Object]"
      />
      <span
        class="label"
      >
        Blog
      </span>
    </span>
  </button>
  <nav
    class="submenu"
  >
    <header
      class="submenuHeader"
    >
      <h2
        class="inverse"
      >
        Read about:
      </h2>
    </header>
    <div
      class="submenuItems"
    >
      <a
        class="submenuItem"
        href="/pwastudio"
      >
        <span
          class="iconContainer"
        >
          <img
            alt="PWA Studio"
            src="[object Object]"
          />
          <span
            class="label"
          >
            PWA Studio
          </span>
        </span>
      </a>
    </div>
    <a
      class="tiertary"
      href="/blog"
      style="margin-top: 30px;"
    >
      Show all
    </a>
  </nav>
</div>
`;
